platform/platform-impl/src/com/intellij/openapi/editor/impl/EditorMarkupModelImpl.java

```diff
@@ -581,8 +575,7 @@ protected boolean isDark() {
 
     @Override
     protected boolean alwaysPaintThumb() {
-      if (scrollbar.getOrientation() == Adjustable.VERTICAL) return !(xcodeLikeScrollbar() && EditorUtil.isRealFileEditor(myEditor));
-      return super.alwaysPaintThumb();
+      return true;
```

Comment:
It is suspicious. Why it returns true, that mean always paint thumb?

Reply:
This method is inside `MyErrorPanel`, to have event listeners bound, this method should return true.

From what I understand it is only used in code editor and only in vertical context, so `scrollbar.getOrientation() == Adjustable.VERTICAL` always evaluates to true, so it always returns result of `return !(xcodeLikeScrollbar() && EditorUtil.isRealFileEditor(myEditor));`.

Under default behaviour `xcodeLikeScrollbar()` evaluates to false, condition is short-circuited and negated, so it returns true.

Under `editor.xcode.like.scrollbar=true` this would return false and hide `MyErrorPanel`, disabling mouse event handlers when thumb is hidden by logic from the parent class that implements OS X scrollbar hiding behaviour.

In the case of this patch, I decided to always render `MyErrorPanel` thumb (which is actually invisible) and have mouse event handlers active, you can see this in action in the demo video linked in PR description where I hover cursor on error stripes.

tl;dr: alwaysPaintThumb returns whether or not to render invisible thumb for error stripes, which under all conditions should be true

Is there a case when this is not desired?